### ISSUE TITLE
[blueprints] One-to-many associations cannot be updated when using the complete object instead of the id

### ISSUE THREAD

**Issue by iit-federico-bozzini** [balderdashy/sails#3396]
Let's say that we have these 2 models:

```
module.exports = {

    attributes: {
        name:'STRING',
        color:'STRING',
        owner:{
            model:'user'
        }
    }

}

module.exports = {

    attributes: {
        name:'STRING',
        age:'INTEGER',
        pets:{
            collection: 'pet',
            via: 'owner'
        }
    }

}
```

If we use the blueprints to GET a specific pet with its owner

    GET /pets/1?populate=owner

then we get an object with the owner field populated with an object.

If we try to PUT this object

    PUT /pets/1

There is an error.

```
{
  "error": "E_VALIDATION",
  "status": 400,
  "summary": "1 attribute is invalid",
  "model": "User",
  "invalidAttributes": {
    "role": [
      {
        "rule": "string",
        "message": "`undefined` should be a string (instead of \"null\", which is a object)"
      },
      {
        "rule": "in",
        "message": "\"in\" validation rule failed for input: null"
      },
      {
        "rule": "required",
        "message": "\"required\" validation rule failed for input: null"
      }
    ]
  }
}
```

A simple solution can be to substituted the populated associations with the respective ids before the PUT request but this looks a little bit "dirty" in my opinion.

**Comment by japel** [comment]
I cannot reproduce your issue with sails 0.11.3
would you provide a repo to test?

**Comment by jdcauley** [comment]
@iit-federico-bozzini 

If you can provide the project and your environment so that we could reproduce the issue please let us know.

Otherwise I will close the issue in a 7 Days.

Jordan

**Comment by sailsbot** [comment]
Thanks for posting, @iit-federico-bozzini.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
